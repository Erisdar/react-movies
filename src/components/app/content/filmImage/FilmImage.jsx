import React from 'react';
import './filmImage.css';

export default ({ logo, alt }) => (<img src={logo} alt={alt} className="film-item__film-image" />);

