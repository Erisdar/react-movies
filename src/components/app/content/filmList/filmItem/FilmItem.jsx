import FilmImage from 'app/content/filmImage/FilmImage';
import ImageErrorBoundary from 'error/ImageErrorBoundary';
import React from 'react';
import { Link } from 'react-router-dom';
import FilmInfo from './filmInfo/FilmInfo';
import './filmItem.css';

export default ({ film, setDetailFilm }) => (
    <Link to={`/film/${film.id}`} onClick={() => setDetailFilm(film)} className="film-item film-list__film-item d-flex flex-column justify-content-start">
        <div className="logo-container film-item__logo-container">
            <ImageErrorBoundary>
                <FilmImage logo={film.poster_path} alt={film.title} />
            </ImageErrorBoundary>
        </div>
        <FilmInfo title={film.title} year={new Date(film.release_date).getFullYear()} genres={film.genres.join(', ')} />
    </Link>
);