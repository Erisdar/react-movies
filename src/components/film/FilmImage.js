import React from 'react';
import '../../styles/film/filmImage.css';

export default ({logo, alt}) => (
    <img src={require(`../../img/${logo}`)} alt={alt} className="film-logo" />
);