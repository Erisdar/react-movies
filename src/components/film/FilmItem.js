import React from 'react';
import '../../styles/film/filmItem.css';
import ImageErrorBoundary from '../error/ImageErrorBoundary';
import FilmImage from './FilmImage';
import FilmInfo from './FilmInfo';

export default ({ film, moreDetail }) => (
    <div className="film-item d-flex flex-column justify-content-start" >
        <div className="logo-container" onClick={() => moreDetail()}>
            <ImageErrorBoundary>
                <FilmImage logo={film.logo} alt={film.title} />
            </ImageErrorBoundary>
        </div>
        <FilmInfo title={film.title} year={film.year} genre={film.genre} />
    </div >
);