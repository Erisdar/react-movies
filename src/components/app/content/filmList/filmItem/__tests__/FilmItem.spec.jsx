import FilmImage from 'app/content/filmImage/FilmImage';
import { mount } from 'enzyme';
import React from 'react';
import FilmItem from '../FilmItem';

jest.mock('app/content/filmImage/FilmImage');

const testFilm = require('data/film.json').find(film => film.id === "1");

describe('Film item component', () => {
    it('should be render correctly with FilmImage', () => {
        FilmImage.mockImplementation(({ logo, alt }) => {
            return <img src={require(`src/img/${logo}`)} alt={alt} className="film-logo" />
        });
        const component = mount(<FilmItem film={testFilm} />);
        expect(component).toMatchSnapshot();
    });

    it('should be call moreDetail()', () => {
        let moreDetail = jest.fn();
        const component = mount(<FilmItem film={testFilm} moreDetail={moreDetail} />);
        component.find('.logo-container').simulate('click');
        expect(moreDetail).toHaveBeenCalled();
    });

    it('should be render correctly with ImageErrorBoundary', () => {
        console.error = jest.fn();
        console.log = jest.fn();
        FilmImage.mockImplementation(() => {
            throw new Error('Test error');
        });
        const component = mount(<FilmItem film={testFilm} />);
        expect(FilmImage).toThrow();
        expect(component).toMatchSnapshot();
    });
});
