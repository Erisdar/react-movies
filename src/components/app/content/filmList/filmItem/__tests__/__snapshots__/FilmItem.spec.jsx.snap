// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film item component should be render correctly with FilmImage 1`] = `
<_default
  film={
    Object {
      "description": "The lead character, called 'The Bride', was a member of the Deadly Viper Assassination Squad, led by her lover 'Bill.' Upon realizing she was pregnant with Bill's child, 'The Bride' decided to escape her life as a killer. She fled to Texas, met a young man, who, on the day of their wedding rehearsal was gunned down by an angry and jealous Bill (with the assistance of the Deadly Viper Assassination Squad). Four years later, 'The Bride' wakes from a coma, and discovers her baby is gone. She, then, decides to seek revenge upon the five people who destroyed her life and killed her baby. The saga of Kill Bill Volume I begins.",
      "duration": "111",
      "genre": "Action & Adventure",
      "id": "1",
      "logo": "kill-bill-v1.jpg",
      "rating": "8.1",
      "title": "Kill Bill vol 1",
      "year": "2003",
    }
  }
>
  <div
    className="film-item film-list__film-item d-flex flex-column justify-content-start"
  >
    <div
      className="logo-container film-item__logo-container"
      onClick={[Function]}
    >
      <ImageErrorBoundary>
        <$default
          alt="Kill Bill vol 1"
          logo="kill-bill-v1.jpg"
        >
          <img
            alt="Kill Bill vol 1"
            className="film-logo"
            src="test-file-stub"
          />
        </$default>
      </ImageErrorBoundary>
    </div>
    <_default
      genre="Action & Adventure"
      title="Kill Bill vol 1"
      year="2003"
    >
      <div
        className="d-flex flex-column h-100 justify-content-between film-info film-item__film-info"
      >
        <div
          className="d-flex justify-content-between align-items-start"
        >
          <p
            className="film-info__title"
            data-cy="film-info-title"
          >
            Kill Bill vol 1
          </p>
          <p
            className="fiml-info__year"
          >
            2003
          </p>
        </div>
        <div
          className="h-50"
        >
          <p
            className="film-info__genre"
            data-cy="film-info-genre"
          >
            Action & Adventure
          </p>
        </div>
      </div>
    </_default>
  </div>
</_default>
`;

exports[`Film item component should be render correctly with ImageErrorBoundary 1`] = `
<_default
  film={
    Object {
      "description": "The lead character, called 'The Bride', was a member of the Deadly Viper Assassination Squad, led by her lover 'Bill.' Upon realizing she was pregnant with Bill's child, 'The Bride' decided to escape her life as a killer. She fled to Texas, met a young man, who, on the day of their wedding rehearsal was gunned down by an angry and jealous Bill (with the assistance of the Deadly Viper Assassination Squad). Four years later, 'The Bride' wakes from a coma, and discovers her baby is gone. She, then, decides to seek revenge upon the five people who destroyed her life and killed her baby. The saga of Kill Bill Volume I begins.",
      "duration": "111",
      "genre": "Action & Adventure",
      "id": "1",
      "logo": "kill-bill-v1.jpg",
      "rating": "8.1",
      "title": "Kill Bill vol 1",
      "year": "2003",
    }
  }
>
  <div
    className="film-item film-list__film-item d-flex flex-column justify-content-start"
  >
    <div
      className="logo-container film-item__logo-container"
      onClick={[Function]}
    >
      <ImageErrorBoundary>
        <img
          className="film-item__film-image"
          src="test-file-stub"
        />
      </ImageErrorBoundary>
    </div>
    <_default
      genre="Action & Adventure"
      title="Kill Bill vol 1"
      year="2003"
    >
      <div
        className="d-flex flex-column h-100 justify-content-between film-info film-item__film-info"
      >
        <div
          className="d-flex justify-content-between align-items-start"
        >
          <p
            className="film-info__title"
            data-cy="film-info-title"
          >
            Kill Bill vol 1
          </p>
          <p
            className="fiml-info__year"
          >
            2003
          </p>
        </div>
        <div
          className="h-50"
        >
          <p
            className="film-info__genre"
            data-cy="film-info-genre"
          >
            Action & Adventure
          </p>
        </div>
      </div>
    </_default>
  </div>
</_default>
`;
